csharp/src/Google.Protobuf/Reflection/Descriptor.cs

```diff
@@ -4912,10 +4914,11 @@ internal enum OptimizeMode {
 
   }
 
-  internal sealed partial class MessageOptions : pb::IExtendableMessage<MessageOptions> {
+  public sealed partial class MessageOptions : pb::IExtendableMessage<MessageOptions> {
     private static readonly pb::MessageParser<MessageOptions> _parser = new pb::MessageParser<MessageOptions>(() => new MessageOptions());
     private pb::UnknownFieldSet _unknownFields;
     internal pb::ExtensionSet<MessageOptions> _extensions;
+    private pb::ExtensionSet<MessageOptions> _Extensions => _extensions;
```

Comment:
qq: why expose a private property `private pb::ExtensionSet<MessageOptions>` for an internal field `internal pb::ExtensionSet<MessageOptions>`?

Reply:
`_Extensions` is there to speed up accessing the set in reflection for `IsInitialized` (building a delegate around a real get method is faster than accessing the field via a reflection call). 

`_extensions` is internal in descriptor protos since it's inner value dictionary is used by `CustomOptions` in an attempt to emulate its past behavior even though it's deprecated with this PR.